### DISCUSSION TITLE
Where should the shuffle button be?

### DISCUSSION THREAD

**Discussion by candroid-man** [z-huang/music#554] [Polls]
# This is InnerTune's mobile player:

<img width=200 src="https://user-images.githubusercontent.com/51019003/216756001-5d1bf779-2e5f-4ea9-8cbf-866657594ff7.png" />

## Notes
Notice the lack of a shuffle button. To get to the shuffle button, you must open the queue by swiping up from the bottom. This can also cause problems if you have a gesture bar because instead of swiping up in InnerTune, you can easily go to the home screen instead.

# This is the Spotify mobile player:
<img width=200 src="https://user-images.githubusercontent.com/51019003/216755929-4a47eca8-6fd5-43b3-b691-02b30eb910f8.png" />

## Notes
Here is Spotify, you have all the buttons that InnerTune has, except the favorite/like button is moved elsewhere on the player. Because of how InnerTune (and YouTube Music) is laid out, the title of the song is centered which I, personally, prefer over Spotify's approach; however, this creates less room for other buttons. Without changing the way InnerTune looks at all, I still think it would be possible to fit the shuffle button on the main player. Perhaps the like button can be put on the bottom bar?

I constantly use the shuffle button, so changing it to be on the player rather than in the queue would be beneficial to me personally. I wanted to see if anyone else had any input on the topic, so:

**Comment by z-huang** [comment]
FYI, I've added the sleep timer button at the bottom of the player, and the button transforms to counter text when sleep timer is started. Also, I plan to limit the number of buttons at the bottom to 5, so it will be hard to move the like button elsewhere.

<img width=200 src="https://user-images.githubusercontent.com/39427017/216769462-bb225ee5-0133-404e-b8cf-b1029d3a725e.jpg" /> <img width=200 src="https://user-images.githubusercontent.com/39427017/216769456-3f65cedc-85b5-44a5-bb5d-7646a479e782.jpg" />

**Reply by owenconnorz** [reply]
Man that's looking so good 👍

**Reply by yant0** [reply]
The current shuffle button on compose debug app have a transparent background which does not have a big constrast on the small grey shuffle icon on bottom left.

IMO, i would like the "add to library" button on the bottom be replaced since there are already 2 ways of having the song in the library:
1. Through the like system
2. The auto add playing songs to library

Since i like not having filler songs, i choose to disable auto add and selectively add each one through the long press / like system because it is a similar system akin to using traditional music players.
